elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -420,11 +422,17 @@ public boolean matchesId(String checkId) {
      */
     public boolean updateAttribute(String fieldName, Object newVal) {
         Type<?> fieldClass = dictionary.getType(getResourceType(), fieldName);
-        newVal =  dictionary.coerce(obj, newVal, fieldName, fieldClass);
+        final Object coercedNewValue = dictionary.coerce(obj, newVal, fieldName, fieldClass);
         Object val = getValueUnchecked(fieldName);
-        checkFieldAwareDeferPermissions(UpdatePermission.class, fieldName, newVal, val);
-        if (!Objects.equals(val, newVal)) {
-            this.setValueChecked(fieldName, newVal);
+        checkFieldAwareDeferPermissions(UpdatePermission.class, fieldName, coercedNewValue, val);
+        if (!Objects.equals(val, coercedNewValue)) {
+            if (val == null
+                    || coercedNewValue == null
+                    || !dictionary.isComplexAttribute(EntityDictionary.getType(obj), fieldName)) {
+                this.setValueChecked(fieldName, newVal);
```

Comment:
Looked into functions: `setValueChecked`, `checkFieldAwareDeferPermissions`, `setValue` and `dictionary.setValue`, I believe that any update/new value for attribute should be coerced before it is passed in to these functions. Am I correct?

Reply:
Yes.  That is correct.